feat(tutorial): add step 5 content (#1389)
carbon-design-system/carbon-for-ibm-dotcom-website: src/pages/developing/web-components-tutorial/step-5.mdx

---

> title: 5. Deploying to Github Pages

Reviewer: ```suggestion
title: 5. Deploying to GitHub Pages
```

<PageDescription>

This step takes what we’ve built so far and prepares the app for a production environment.

> We’ll be deploying the production build to Github Pages.

Reviewer: ```suggestion
We’ll be deploying the production build to GitHub Pages.
```

</PageDescription>

<AnchorLinks>

<AnchorLink>Fork, clone, and branch</AnchorLink>
<AnchorLink>Build for production</AnchorLink>

> <AnchorLink>Deploying to Github Pages</AnchorLink>

Reviewer: ```suggestion
<AnchorLink>Deploying to GitHub Pages</AnchorLink>
```

<AnchorLink>Submit pull request</AnchorLink>

</AnchorLinks>

## Preview

> A [preview]() of what you’ll build (visually no different, but built for production):

Reviewer: ```suggestion
Here's a [preview]() of what you’ll build. It's not different visually, but is now built for production.
```

<Preview
  height="400"

>   title="Carbon for IBM.com Tutorial Step 5"

Reviewer: ```suggestion
  title="Carbon for IBM.com tutorial step 5"
```

```bash
git fetch upstream
git checkout -b web-components-step-5 upstream/web-components-step-5
```

### Build and start app

> Install the app’s dependencies (in case you’re starting fresh in your current directory and not continuing from the previous step):

Reviewer: ```suggestion
If you’re starting fresh in your current directory and not continuing from the previous step, install the app's dependencies:
```

```bash
yarn install
```

## Build for production

Before we deploy our app, we need to create an optimized production build with this command.

Run in your terminal

```bash
yarn build
```

Looking at `package.json`, you’ll find `yarn build` runs the `webpack-prod` configurations. This builds the app for production to the `dist` folder.
It creates minified bundles, lighter weight source maps, and optimizes assets in production mode to improve load time.

## Deploying to Github Pages

Next we'll be pushing the generated `dist` folder to [Github Pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)
which allows you to host your files directly from your repository in Github.

To do this, we'll first commit our `dist` folder so that git know we have a subtree (subfolder with our site).

```bash
git add dist && git commit -m "chore(deploy): publish build files"
```

Then we'll use `subtree push` to push to the `gh-pages` branch on Github.

```bash
git subtree push --prefix dist origin gh-pages
```

> To have Github Pages point to the `gh-pages` branch, go to `github.com` to your forked Carbon for IBM.com Web Components tutorial repository. Go to your "Settings" and under the "Pages" tab, specify the branch, `gh-pages`,

Reviewer: ```suggestion
To have GitHub Pages point to the `gh-pages` branch, go to `github.com` to your forked Carbon for IBM.com Web Components tutorial repository. Go to your "Settings" and under the "Pages" tab, specify the branch, `gh-pages`,
```

as your source. Once you click save, you should be able to see your site published at the `*.github.io/carbon-for-ibm-dotcom-tutorial/` link shown on the page.

![Screenshot of Github settings dashboard](../../../images/tutorial/deploy-on-github-pages.png)

> Congratulations! Your work is now hosted on Github Pages!

Reviewer: ```suggestion
Congratulations! Your work is now hosted on GitHub Pages!
```
